src/consensus/consensus.h

```diff
@@ -6,10 +6,26 @@
 #ifndef BITCOIN_CONSENSUS_CONSENSUS_H
 #define BITCOIN_CONSENSUS_CONSENSUS_H
 
+#include <stdint.h>
+
+static const uint64_t BIP102_FORK_TIME = 1462406400; // May 5 2016, midnight UTC
+
 /** The maximum allowed size for a serialized block, in bytes (network rule) */
-static const unsigned int MAX_BLOCK_SIZE = 1000000;
+inline unsigned int MaxBlockSize(uint64_t nTime) {
```

Comment:
Should use median time or height so that you can always predict whether the next block needs to follow the new rules or not.